occurrences/tests/test_models.py

```diff
@@ -172,6 +172,65 @@ def test_study_group_with_enrolment_count_by_name(
     assert the_school_2nd_instance in groups_with_seven_enrolments
 
 
+@pytest.mark.django_db
+def test_study_group_with_organisation_ids(mock_get_place_data, mock_get_event_data):
+    EnrolmentFactory.create_batch(10)
+    groups = StudyGroup.objects.all()
+
+    # Without annotation
+    for group in groups:
+        assert hasattr(group, "organisation_ids") is False
+
+    groups = StudyGroup.objects.with_organisation_ids().all()
+
+    # With annotation
+    for group in groups:
+        assert hasattr(group, "organisation_ids") is True
+        assert group.organisation_ids is not None
+
+    group_without_enrolment = StudyGroupFactory()
+
+    # Without enrolment, so without organisation
+    StudyGroup.objects.with_organisation_ids().get(
+        id=group_without_enrolment.id
+    ).organisation_ids is None
+
+    # With multiple enrolments to multiple organisations
+    group_with_multiple_organisations = StudyGroupFactory()
+    enrolments = EnrolmentFactory.create_batch(
+        2, study_group=group_with_multiple_organisations
+    )
+    enrolments_organisation_ids = [
+        enrolment.occurrence.p_event.organisation.id for enrolment in enrolments
+    ]
+    group_organisation_ids = (
+        StudyGroup.objects.with_organisation_ids()
+        .get(id=group_with_multiple_organisations.id)
+        .organisation_ids
+    )
+    assert len(group_organisation_ids) == 2
+    assert all(
+        (org_id in enrolments_organisation_ids) for org_id in group_organisation_ids
+    )
+
+
+@pytest.mark.django_db
+def test_study_group_user_can_view(
+    mock_get_place_data, mock_get_event_data, organisation
+):
+    person = PersonFactory(user=UserFactory())
+    person.organisations.add(organisation)
+    organisation_enrolments = EnrolmentFactory.create_batch(
+        5, occurrence__p_event__organisation=organisation
+    )
+    organisation_groups = [
+        enrolment.study_group for enrolment in organisation_enrolments
+    ]
+    StudyGroupFactory.create_batch(5)
+    StudyGroup.objects.all().count() == 10
+    StudyGroup.objects.user_can_view(person.user).count() == len(organisation_groups)
```

Comment:
asserts? (See Sonarcloud [report](https://sonarcloud.io/project/issues?resolved=false&types=BUG&sinceLeakPeriod=true&pullRequest=311&id=City-of-Helsinki_kultus-api))